src/backend/aspen/app/views/usergroup.py

```diff
@@ -1,17 +1,40 @@
-from flask import jsonify, session
+import json
+from typing import Dict, Union
+
+from flask import jsonify, request, Response, session
 
 from aspen.app.app import application, requires_auth
 from aspen.app.views.api_utils import get_usergroup_query
 from aspen.database.connection import session_scope
+from aspen.database.models.usergroup import User
 
 
-@application.route("/api/usergroup", methods=["GET"])
+@application.route("/api/usergroup", methods=["GET", "PUT"])
 @requires_auth
 def usergroup():
     with session_scope(application.DATABASE_INTERFACE) as db_session:
         # since authentication is required to access view this information is guaranteed
         # to be in session
-        profile = session["profile"]
-        user = get_usergroup_query(db_session, profile["user_id"]).one()
-        user_groups = {"user": user.to_dict(), "group": user.group.to_dict()}
-        return jsonify(user_groups)
+        profile: Dict[str, str] = session["profile"]
+        user: User = get_usergroup_query(db_session, profile["user_id"]).one()
+
+        if request.method == "GET":
+            user_groups: Dict[str, Dict[str, Union[str, bool]]] = {
+                "user": user.to_dict(),
+                "group": user.group.to_dict(),
+            }
+            return jsonify(user_groups)
+
+        if request.method == "PUT":
+            fields_to_update: Dict[str, Union[str, bool]] = json.loads(
+                request.get_json()
+            )
+            for key, value in fields_to_update.items():
```

Comment:
Should we restrict what fields a user is allowed to update using an allow list? Or are the fields here pretty harmless even if they change things unexpectedly? For example, if they change group id to something else, would they gain access to something they shouldn't? Thank you!

Reply:
oh that's a good idea!! thanks!

Reply:
@tihuan just added this, let me know if you think that solves it!

Reply:
Awesomeee! Thanks so much for the quick update, @phoenixAja 🎉 ! LGTM!!